functable.c

```diff
@@ -16,310 +16,219 @@ static void init_functable(void) {
+        ft.crc32_fold_copy = &crc32_fold_pclmulqdq_copy;
+        ft.crc32_fold_final = &crc32_fold_pclmulqdq_final;
+    }
 #endif
-#ifdef POWER8_VSX_ADLER32
-    if (power_cpu_has_arch_2_07)
-        ft.adler32 = &adler32_power8;
+    // X86 - AVX
+#ifdef X86_AVX_CHUNKSET
+    if (x86_cpu_has_avx2) {
+        ft.chunksize = &chunksize_avx;
+        ft.chunkmemset_safe = &chunkmemset_safe_avx;
+        ft.inflate_fast = &inflate_fast_avx;
+    }
 #endif
-
-    // adler32_fold_copy_stub:
-    ft.adler32_fold_copy = &adler32_fold_copy_c;
-#ifdef X86_SSE42_ADLER32
-    if (x86_cpu_has_sse42)
-        ft.adler32_fold_copy = &adler32_fold_copy_sse42;
+#ifdef X86_AVX2
+    if (x86_cpu_has_avx2)
+        ft.slide_hash = &slide_hash_avx2;
+#endif
+#if defined(X86_AVX2) && defined(HAVE_BUILTIN_CTZ)
+    if (x86_cpu_has_avx2) {
+        ft.longest_match = &longest_match_avx2;
+        ft.longest_match_slow = &longest_match_slow_avx2;
+        ft.compare256 = &compare256_avx2;
+    }
 #endif
 #ifdef X86_AVX2_ADLER32
-    if (x86_cpu_has_avx2)
+    if (x86_cpu_has_avx2) {
+        ft.adler32 = &adler32_avx2;
         ft.adler32_fold_copy = &adler32_fold_copy_avx2;
+    }
 #endif
 #ifdef X86_AVX512_ADLER32
-    if (x86_cpu_has_avx512)
+    if (x86_cpu_has_avx512) {
+        ft.adler32 = &adler32_avx512;
         ft.adler32_fold_copy = &adler32_fold_copy_avx512;
+    }
 #endif
 #ifdef X86_AVX512VNNI_ADLER32
-    if (x86_cpu_has_avx512vnni)
+    if (x86_cpu_has_avx512vnni) {
+        ft.adler32 = &adler32_avx512_vnni;
         ft.adler32_fold_copy = &adler32_fold_copy_avx512_vnni;
+    }
 #endif
 
-    // crc32_fold_reset_stub:
-    ft.crc32_fold_reset = &crc32_fold_reset_c;
-#ifdef X86_PCLMULQDQ_CRC
-    if (x86_cpu_has_pclmulqdq)
-        ft.crc32_fold_reset = &crc32_fold_pclmulqdq_reset;
-#endif
-
-    // crc32_fold_copy_stub:
-    ft.crc32_fold_copy = &crc32_fold_copy_c;
-#ifdef X86_PCLMULQDQ_CRC
-    if (x86_cpu_has_pclmulqdq)
-        ft.crc32_fold_copy = &crc32_fold_pclmulqdq_copy;
-#endif
-
-    // crc32_fold_stub:
-    ft.crc32_fold = &crc32_fold_c;
-#ifdef X86_PCLMULQDQ_CRC
-    if (x86_cpu_has_pclmulqdq)
-        ft.crc32_fold = &crc32_fold_pclmulqdq;
-#endif
 
-    // crc32_fold_final_stub:
-    ft.crc32_fold_final = &crc32_fold_final_c;
-#ifdef X86_PCLMULQDQ_CRC
-    if (x86_cpu_has_pclmulqdq)
-        ft.crc32_fold_final = &crc32_fold_pclmulqdq_final;
-#endif
-
-    //chunksize_stub:
-    ft.chunksize = &chunksize_c;
-#ifdef X86_SSE2_CHUNKSET
-# if !defined(__x86_64__) && !defined(_M_X64) && !defined(X86_NOCHECK_SSE2)
-    if (x86_cpu_has_sse2)
-# endif
-        ft.chunksize = &chunksize_sse2;
-#endif
-#ifdef X86_AVX_CHUNKSET
-    if (x86_cpu_has_avx2)
-        ft.chunksize = &chunksize_avx;
+    // ARM - NEON
+#if defined(ARM_NEON) && defined(HAVE_BUILTIN_CTZLL)
+    if (arm_cpu_has_neon) {
+        ft.longest_match = &longest_match_neon;
+        ft.longest_match_slow = &longest_match_slow_neon;
```

Comment:
Are we missing `compare256_neon` here?

Reply:
@nmoinvaz It sure looks like it, but it was already missing for some reason.. 🤯